qiskit/assembler/assemble_schedules.py

```diff
@@ -246,19 +246,28 @@ def _validate_meas_map(instruction_map: Dict[Tuple[int, instructions.Acquire],
     Raises:
         QiskitError: If the instructions do not satisfy the measurement map.
     """
+    sorted_inst_map = sorted(instruction_map.items(), key=lambda item: item[0])
     meas_map_sets = [set(m) for m in meas_map]
-
-    # Check each acquisition time individually
-    for _, instrs in instruction_map.items():
+    all_measured_qubits = [inst.channel.index for _, insts in sorted_inst_map
+                           for inst in insts]
+
+    # 1. if the qubits are not in the meas_map -- Raise Error
+    if not set(all_measured_qubits).issubset(set.union(*meas_map_sets)):
+        raise QiskitError("The measured qubits - {} is not in the measurement "
+                          "map - {}".format(all_measured_qubits, meas_map))
+    # 2. if there is time overlap:
+    #    - if the overlap is in the same meas_map -- Raise Error
+    for idx, instr in enumerate(sorted_inst_map[:-1]):
+        if sorted_inst_map[idx+1][0] == instr[0]:
+            continue
         measured_qubits = set()
-        for inst in instrs:
-            measured_qubits.add(inst.channel.index)
-
-        for meas_set in meas_map_sets:
-            intersection = measured_qubits.intersection(meas_set)
-            if intersection and intersection != meas_set:
-                raise QiskitError('Qubits to be acquired: {} do not satisfy required qubits '
-                                  'in measurement map: {}'.format(measured_qubits, meas_set))
+        measured_qubits.update([inst.channel.index for inst in instr[1]],
+                               [inst.channel.index for inst in sorted_inst_map[idx+1][1]])
+        inst_end_time = instr[0][0] + instr[0][1]
```

Comment:
I'm not sure why this is double indexed -- I feel like it should raise an error. Don't we have this:
`instr = (start_time, duration)`?